### DISCUSSION TITLE
is that possible to input but not explain a feature using KernelExplainer?

### DISCUSSION THREAD

**Discussion by kwtk86** [shap/shap#2954] [Q&A]
i m using KernelExplainer to explain my ANN model. nine features are the inputs of the ANN model.
It is easy to explain the influences of all the features using KernelExplainer. However, i only want eight of the features to be explained.
For example, the nine input features are (feature1-9), i hope only the feature1-8 to be explained
is it possible to do that?

**Comment by jainraj** [comment]
That is not possible through any explainer as SHAP assumes that all features participate in the model predictions (i.e., all players contribute). 

What is possible though is you can change the background dataset to remove the effect of that variable. Relevant papers which discuss this sort of usecase: 

1. [The Explanation Game: Explaining Machine Learning Models Using Shapley Values - Merrick and Taly](https://arxiv.org/pdf/1909.08128.pdf)
2. [Explaining a Series of Models by Propagating Shapley Values - Chen et al.](https://arxiv.org/pdf/2105.00108.pdf) - Section 3.2
